// ads-client/dist/client-error.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
class ClientError extends Error {
    constructor(message, adsError) {
        super(message);
        this.trace = [];
        Object.setPrototypeOf(this, new.target.prototype);
        this.name = "ClientError";
        message && this.trace.push(message);
        if (adsError !== undefined && adsError.stack) {
            //console.log("moro:", this.stack)
            //process.exit()
            this.stack = this.stack + "\n" + adsError.stack; //+ this.stack?.split('\n').slice(0, 2).join('\n') + '\n';
        }
        if (adsError instanceof ClientError) {
            this.adsError = adsError.adsError;
            this.trace = this.trace.concat(adsError.trace);
        }
        else if (!(adsError instanceof Error) && adsError !== undefined) {
            const error = adsError;
            this.adsError = {
                errorCode: error.errorCode,
                errorStr: error.errorStr
            };
        }
        Object.setPrototypeOf(this, ClientError.prototype);
    }
}
exports.default = ClientError;
